Common Utilities/ConfigObjects/PlayerUpgradeChance.cs

```diff
@@ -7,14 +7,14 @@ public class PlayerUpgradeChance
         public RoleTypeId Original { get; set; }
         public RoleTypeId New { get; set; }
         public double Chance { get; set; }
-        public bool KeepInventory { get; set; } = true;
+        public RoleSpawnFlags SpawnFlags { get; set; }
```

Comment:
you just broke every single server's config file, good job.

Reply:
The UpgradeChance is only used for 914 player upgrades.
This means their spawn location will either be the 914 output, or 914 teleport position.
This means we can't allow UseSpawnpoint flag.
That means RoleSpawnFlags.AssignInventory and RoleSpawnFlags.None are the only options.
Which are already handled by the existing bool.

There's no reason for this change, as it doesn't enable any additional features/changes that don't exist already.

And if we for some reason do want to allow 914 class changing to use spawn points for the spawn location, it's better to add a ``public bool UseRoleSpawnPoint`` value that defaults to false. This will get added to existing configs but not change the behavior unless the owner wants it to be changed.

Removing the existing property will, however, cause Yaml to be unable to serialize the config into this object, thus resulting in an exception, which may or may not cause Yaml to completely reset the config file, wiping the server's config values.